graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -248,17 +248,20 @@ def _log(self, job_id: str, future: "Future[Any]", database: Optional[str] = Non
                     continue
 
             progress_percent = progress["progress"][0]
-            if not progress_percent == "n/a":
-                task_name = progress["taskName"][0].split("|--")[-1][1:]
-                pbar = pbar or tqdm(total=100, unit="%", desc=task_name)
-            else:
+            if progress_percent == "n/a":
                 return
 
+            task_name = progress["taskName"][0].split("|--")[-1][1:]
+            if task_name not in pbars:
+                pbars[task_name] = tqdm(total=100, unit="%", desc=task_name)
+            pbar = pbars[task_name]
```

Comment:
Why do we need this dictionary? Don't we only have one progress bar?

Reply:
I thought its needed as my first assumption was result ordering, but in the end its not as we only retrieve the root task.

I added a `LIMIT 1` in the cypher query to make this clearer.